dietpi/dietpi-software

```diff
@@ -4136,6 +4143,103 @@ _EOF_
+			local micro_systemd="/etc/systemd/system/$micro_name.service"
+			local micro_functions="/etc/bashrc.d/$micro_name.sh"
+
+			# User
+			Create_User -d "$micro_data_dir" "$micro_name"
+
+			# 1. Create a Python 3.10 environment via pyenv
+			# - https://python-poetry.org/docs/managing-environments/
+			Download_Install 'https://github.com/pyenv/pyenv/archive/master.tar.gz'
+			# - Start with fresh instance to allow clean pyenv and Python updates and fix broken instances.
+			[[ -d $micro_pyenv_dir ]] && G_EXEC rm -R "$micro_pyenv_dir"
+			G_EXEC mv pyenv-master "$micro_pyenv_dir"
+
+			# 2. Clone the repo into data dir
+			[[ -d $micro_data_dir ]] || G_EXEC_OUTPUT=1 G_EXEC git clone 'https://git.sr.ht/~tsileo/microblog.pub' "$micro_data_dir"
+			G_EXEC chown -R "$micro_name:$micro_name" "$micro_data_dir"
+
+			# 3. Generate script to activate pyenv
+			echo "#!/bin/dash
+cd $micro_pyenv_dir
+export PYENV_ROOT='$micro_pyenv_dir'
+export PATH=\"\$PYENV_ROOT/bin:\$PATH\"
+eval \"\$(pyenv init --path)\"
+eval \"\$(pyenv init -)\"
+cd $micro_data_dir
+PS1=\"(microblog.pub) \$PS1\"" > "$micro_pyenv_activate_file"
+
+			# 4. Install Python 3.10 and Poetry
+			G_EXEC chown -R "$micro_name:$micro_name" "$micro_pyenv_dir"
+			G_EXEC_DESC='Installing Python with Poetry into pyenv'
+			G_EXEC_OUTPUT=1 G_EXEC sudo -u "$micro_name" dash -c "
+source $micro_pyenv_activate_file
+pyenv install $micro_python_version
+pyenv local $micro_python_version
+pip3 install -U pip setuptools wheel
+pip3 install poetry
+poetry install
+# Skip these for now:
+#poetry run inv configuration-wizard
+#poetry run inv migrate-db
+poetry env info | grep -oP '(?<=^Path:\s).*/virtualenvs/.*' > '$micro_virtualenv'"
+
+			G_EXEC_DESC='Checking for Poetry virtualenv' G_EXEC eval "[[ -s $micro_virtualenv ]]"
+
+			# 5. Generate functions for later configuration and install to profile
+			cat << _EOF_ > "$micro_functions"
+#!/bin/dash
+microblog_pub()
+{
+	case \$1 in
+		v*) bash --init-file '$micro_pyenv_activate_file';; # venv
+		c*) dash -c 'source $micro_pyenv_activate_file; poetry run inv configuration-wizard && poetry run inv migrate-db';; # configure
+		u*) dash -c 'source $micro_pyenv_activate_file; git pull && poetry run inv update';; # update
+		*) echo 'microblog.pub helper script
+
+usage: microblog_pub <venv|configure|update>
+
+where:
+  configure   runs the configuration wizard and updates the microblog environment.
+  update      pulls the latest changes from the microblog repository and reconfigures.
+  venv        starts a virtual environment where the microblog can be configured.' && return 1;;
+	esac
+
+sudo systemctl restart $micro_name
+}
+_EOF_
+			# 5. Generate systemd script
+			cat << _EOF_ > "$micro_systemd"
+[Unit]
+Description=microblog.pub (DietPi)
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+User=$micro_name
+SyslogIdentifier=microblog.pub
+WorkingDirectory=$micro_data_dir
+Environment=VENV_DIR=$micro_virtualenv
```

Comment:
Shouldn't it actually be a directory path here instead of a path to a file which contains the directory path?

Reply:
I broke it myself, just fixed